src/app/clusters/localization-configuration-server/localization-configuration-logic.h

```
@@ -0,0 +1,76 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/AttributeValueEncoder.h>
+#include <app/persistence/String.h>
+#include <lib/support/Span.h>
+#include <platform/DeviceInfoProvider.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class LocalizationConfigurationServerLogic
+{
+public:
+    LocalizationConfigurationServerLogic(DeviceLayer::DeviceInfoProvider & aDeviceInfoProvider) :
+        mDeviceInfoProvider(aDeviceInfoProvider)
+    {}
+
+    /*
+     * Set the active locale.
+     *
+     * @param activeLocale The active locale to set.
+     * @return InteractionModel::Status::Success on success, InteractionModel::Status::ConstraintError if the locale is not valid or
+     * supportedLocale.
+     */
+    Protocols::InteractionModel::Status SetActiveLocale(CharSpan activeLocale);
+
+    /*
+     * Get the active locale.
+     *
+     * @return The active locale.
+     */
+    CharSpan GetActiveLocale();
+
+    /*
+     * Read the supported locales.
+     *
+     * @param aEncoder The encoder to write the supported locales to.
+     * @return CHIP_NO_ERROR on success.
+     */
+    CHIP_ERROR ReadSupportedLocales(AttributeValueEncoder & aEncoder);
+
+    /*
+     * Check if a locale is supported.
+     *
+     * @param newLangTag The locale to check.
+     * @return true if the locale is supported, false otherwise.
+     */
+    bool IsSupportedLocale(CharSpan newLangTag);
+
+private:
+    Storage::String<35> mActiveLocale;
```

we use a String<35> in the attribute write as well. Can we re-use this (maybe make the setter take in the attribute storage as well to persist)? maybe the 35 should be a named constant.

We seem to also want some default logic to auto-persist the value:
  - ember startup reads the default from ember RAM
  - ::Startup() on the cluster starts having references to persistence. So in that case we need to either load from persistent storage or if that fails, use the default and store

Current code seems to do things a bit differently and I am not sure if the flow is correct:
  - ember startup reads value and sets the default (but it is never saved)
  - startup also reads from persistence but ignores errors and calls a set ... this seems odd.
  
Could we verify that we tested (even if manually) these paths and note that in the PR description? The startup logic seems to need some cleanup for clarity.